service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderController.java

```diff
@@ -121,4 +129,17 @@ private Document getDocument(String authorization,
         return uploadDocumentService.uploadPDF(userId, authorization, document.getBytes(),
             C21.getDocumentTitle());
     }
+
+    private String concatGatewayConfigurationUrlAndMostRecentUploadedC21DocumentPath(
+        final String mostRecentUploadedC21DocumentUrl) {
+        final String gatewayUrl = gatewayConfiguration.getGatewayUrl();
+
+        try {
+            URI uri = new URI(mostRecentUploadedC21DocumentUrl);
+            return gatewayUrl + uri.getPath();
+        } catch (URISyntaxException e) {
+            log.error(mostRecentUploadedC21DocumentUrl + " url incorrect.", e);
+        }
+        return "";
+    }
```

Comment:
I am not sure if the controller is however the good place for this logic? maybe could be done in notification content provider? This is the place which really needs to transform the URL, on the controller level it's ok, I think?

Reply:
Would be easier to test as well - and this piece of code should be unit tested.